### ISSUE TITLE
IDISP004 from ReactiveUI's DisposeWith(CompositeDisposable compositeDisposable) mixin in method chains

### ISSUE THREAD

**Issue by PatricioVidal** [DotNetAnalyzers/IDisposableAnalyzers#298]
This is related to #228.

When `DisposeWith()` is used in a method chain the extension still triggers `IDISP004` warning:

```csharp
this.StartCancel
  .ThrownExceptions
  .Subscribe(e => Log.Error(e, $"Error executing '{nameof(this.StartCancel)}' command: {e.Message}"))
  .DisposeWith(disposables);
```

Using a temporary variable to hold the `IDisposable` object does not triggers `IDISP004`:

```csharp
var subscription = this.StartCancel
  .ThrownExceptions
  .Subscribe(e => Log.Error(e, $"Error executing '{nameof(this.StartCancel)}' command: {e.Message}"));

 subscription.DisposeWith(disposables);
```

**Comment by JohanLarsson** [comment]
This was an annoying regression, sorry about taking forever with the fix. Keep the issues coming.
